lib/renderer.js

```diff
@@ -87,6 +88,7 @@ class Renderer extends MarkedRenderer {
 
     if (!parse(href).hostname && !options.config.relative_link
       && options.prependRoot) {
+      if (options.path) href = join(options.path, href);
```

Comment:
There is a possible edge case, this could conflict with embedding an internal non-post-asset image.

for example, when a post "2020/01/02/bar" embeds an image "source/brand/foo.jpg" using `![](/brand/foo.jpg)`, that embed will renders to `<img src="/2020/01/02/bar/brand/foo.jpg">`.

Reply:
Another workaround is having an `ignore:` option, so in previous example, `/brand/**` can be ignored.

Reply:
Another workaround is to use `PostAsset` as used in [asset_img.js](https://github.com/hexojs/hexo/blob/d96c65b865aaa43dd21c2d1bb1e5e8b0ca49d269/lib/plugins/tag/asset_img.js#L21) tag plugin.